src/test/groovy/pl/futurecollars/invoicing/service/TaxCalculatorServiceTest.groovy

```diff
@@ -0,0 +1,120 @@
+package pl.futurecollars.invoicing.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import pl.futurecollars.invoicing.db.Database
+import pl.futurecollars.invoicing.fixtures.InvoiceFixture
+import pl.futurecollars.invoicing.model.Invoice
+import pl.futurecollars.invoicing.utils.JsonService
+import spock.lang.Shared
+import spock.lang.Specification
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class TaxCalculatorServiceTest extends Specification {
+
+    @Autowired
+    private MockMvc mockMvc
```

Comment:
I am not sure if you need MockMvc to test Service class, we usually use MockMvc to test controllers

Reply:
corrected